lib/mysql.js

```diff
@@ -72,31 +72,37 @@ function MySQL(settings) {
     s.connectionLimit = 10;
   }
 
-  var options = {
-    host: s.host || s.hostname || 'localhost',
-    port: s.port || 3306,
-    user: s.username || s.user,
-    password: s.password,
-    timezone: s.timezone,
-    socketPath: s.socketPath,
-    charset: s.collation.toUpperCase(), // Correct by docs despite seeming odd.
-    supportBigNumbers: s.supportBigNumbers,
-    connectionLimit: s.connectionLimit
-  };
-
-  // Don't configure the DB if the pool can be used for multiple DBs
-  if (!s.createDatabase) {
-    options.database = s.database;
-  }
-
-  // Take other options for mysql driver
-  // See https://github.com/strongloop/loopback-connector-mysql/issues/46
-  for (var p in s) {
-    if (p === 'database' && s.createDatabase) {
-      continue;
+  var options;
+  if (s.url) {
+    // use url to override other settings if url provided
+    options = s.url;
+  } else {
+    options = {
+      host: s.host || s.hostname || 'localhost',
+      port: s.port || 3306,
+      user: s.username || s.user,
+      password: s.password,
+      timezone: s.timezone,
+      socketPath: s.socketPath,
+      charset: s.collation.toUpperCase(), // Correct by docs despite seeming odd.
+      supportBigNumbers: s.supportBigNumbers,
+      connectionLimit: s.connectionLimit
+    };
+
+    // Don't configure the DB if the pool can be used for multiple DBs
+    if (!s.createDatabase) {
+      options.database = s.database;
```

Comment:
Are you sure this makes sense when `options` is a string as provided by `s.url`? The same question applies to L100-105.